unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmotesCustomizationComponentController.cs

```diff
@@ -0,0 +1,442 @@
+
+        public void Initialize(UserProfile userProfile, BaseDictionary<string, WearableItem> catalog)
+        {
+            LoadEquippedEmotes();
+            ConfigureView();
+            ConfigureUserProfileAndCatalog(userProfile, catalog);
+            ConfigureShortcuts();
+
+            emotesCustomizationDataStore.isInitialized.Set(view.viewTransform);
+        }
+
+        public void Dispose()
+        {
+            view.onEmoteEquipped -= OnEmoteEquipped;
+            view.onEmoteUnequipped -= OnEmoteUnequipped;
+            view.onSellEmoteClicked -= OnSellEmoteClicked;
+            view.onSlotSelected -= OnSlotSelected;
+            isStarMenuOpen.OnChange -= IsStarMenuOpenChanged;
+            avatarEditorVisible.OnChange -= OnAvatarEditorVisibleChanged;
+            emotesCustomizationDataStore.avatarHasBeenSaved.OnChange -= OnAvatarHasBeenSavedChanged;
+            catalog.OnAdded -= AddEmote;
+            catalog.OnRemoved -= RemoveEmote;
+            emoteAnimations.OnAdded -= OnAnimationAdded;
+            userProfile.OnInventorySet -= OnUserProfileInventorySet;
+            userProfile.OnUpdate -= OnUserProfileUpdated;
+            equipInputAction.OnFinished -= OnEquipInputActionTriggered;
+            showInfoInputAction.OnFinished -= OnShowInfoInputActionTriggered;
+            shortcut0InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut1InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut2InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut3InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut4InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut5InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut6InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut7InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut8InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+            shortcut9InputAction.OnTriggered -= OnNumericShortcutInputActionTriggered;
+        }
+
+        internal void LoadEquippedEmotes()
+        {
+            List<string> storedEquippedEmotes;
+
+            try
+            {
+                storedEquippedEmotes = JsonConvert.DeserializeObject<List<string>>(PlayerPrefsUtils.GetString(PLAYER_PREFS_EQUIPPED_EMOTES_KEY));
+            }
+            catch
+            {
+                storedEquippedEmotes = null;
+            }
+
+            if (storedEquippedEmotes == null)
+                storedEquippedEmotes = GetDefaultEmotes();
+
+            foreach (string emoteId in storedEquippedEmotes)
+            {
+                if (string.IsNullOrEmpty(emoteId))
+                    continue;
+
+                CatalogController.RequestWearable(emoteId);
```

Comment:
perhaps we can leave this as a technical debt with a TODO comment, but we should start avoiding static calls, `CatalogController` is an example that we should create an injectable interface for this behaviour